site/src/components/input.md

```diff
@@ -12,76 +12,102 @@ classPrefixes:
 
 Input elements are used to accept data from the user.
 
-| Class    | Default                             | Disabled                                     |
-| -------- | ----------------------------------- | -------------------------------------------- |
-| `.input` | <input class="input" type="text" /> | <input class="input" type="text" disabled /> |
-
-```html
-<input class="input" type="text" />
-```
+| Class    | Default                                                        | Disabled                                                                 |
+| -------- | -------------------------------------------------------------- | ------------------------------------------------------------------------ |
+| `.input` | <input class="input" type="text" aria-label="Default input" /> | <input class="input" type="text" disabled aria-label="disabled input" /> |
 
 ## Usage
 
 Here is an example of a standard usage of input paired with a label.
 
 ```html
-<label class="label margin-bottom-xxs" for="input-demo">Label</label>
-<input class="input" id="input-demo" type="text" />
+<form action="javascript: void(0)" aria-label="Input example form">
+	<label class="label margin-bottom-xxs" for="input-demo" id="input-demo-label">Label</label>
+	<input class="input" id="input-demo" type="text" aria-labelledby="input-demo-label" />
+</form>
 ```
 
 ### Sizes
 
 `input-sm`/`input-lg` makes input a little bigger or smaller than the default.
 
 ```html
-<input class="input input-sm" type="text" />
-<input class="input margin-top-xs" type="text" />
-<input class="input input-lg margin-top-xs" type="text" />
+<form action="javascript: void(0)" aria-label="Input example form for sizes">
+	<input class="input input-sm" type="text" placeholder="Small input" aria-label="Small input" />
+	<input
+		class="input margin-top-xs"
+		type="text"
+		placeholder="Default input"
+		aria-label="Default input"
+	/>
+	<input
+		class="input input-lg margin-top-xs"
+		type="text"
+		placeholder="Large input"
+		aria-label="Large input"
+	/>
```

Comment:
Musing aloud here, please feel free to resolve as you see fit.

Weirdly enough, I'm a touch concerned about adding things like `aria-label`s to the visible code snippet, because it feels a bit like we're endorsing it as a technique as opposed to, say, recommending `label`...

...but the markup we were showing was incomplete, both for better and for worse.

Hmm...

Reply:
Put in a message in these files: 

> _Note: `aria-labels` are added in the examples below because we are giving partial markup. Consider using a visible label when creating a real experience._